### PR TITLE
New PyCharm Cheat Sheet

### PR DESCRIPTION
**Pull request by Khakhulin**
Instant Answer Page: https://duck.co/ia/view/pychram_cheat_sheet

### CODE REVIEW

**Comment by daxtheduck** [comment]
Metadata placeholder post
            
*If you haven't already, please edit your first comment to include a link to the IA page at the bottom.*
            
---
            
**How to find your IA page**
- New Instant Answer
  -- follow the steps [here](https://duck.co/ia/new_ia) to create your IA page.
              
- Existing Instant Answer
  -- find your IA page from this [list](https://duck.co/ia)

**Comment by pjhampton** [comment]
@Khakhulin hey there!
Can you please update your PR message with the IA page?

File: share/goodie/cheat_sheets/json/pycharm.json [hunk 1]
```diff
@@ -0,0 +1,220 @@
+{
+    "id": "pycharm_cheat_sheet",
+    "name": "PyCharm",
+    "description": "Keyboard shortcuts for PyCharm. It's a python IDE, developed by JetBrains",
```
Comment (adityasiwan) [inline share/goodie/cheat_sheets/json/pycharm.json]: Let's add a period at the end.

File: share/goodie/cheat_sheets/json/pycharm.json [hunk 2]
```diff
@@ -0,0 +1,220 @@
...
+                "val": "Go to previous editor tab"
+            },{
+                "key": "[Ctrl] [G]",
+                "val": "Go to line"
+            },{
+                "key": "[Ctrl] [E]",
+                "val": "Recent files popup"
+            },{
+                "key": "[Ctrl] [Shift] [H]",
+                "val": "method hierarchy"
```
Comment (adityasiwan) [inline share/goodie/cheat_sheets/json/pycharm.json]: Let's change `method` to `Method` 😄

File: share/goodie/cheat_sheets/json/pycharm.json [hunk 3]
```diff
@@ -0,0 +1,220 @@
...
+        "Refactoring",
+        "VCS"
+    ],
+
+    "sections": {
+        "General": [
+            {
+                "key": "[Ctrl] [S]",
+                "val": "Save all"
+            },{
```
Comment (ManrajGrover) [inline share/goodie/cheat_sheets/json/pycharm.json]: Let's add a space after `,`.

Comment (ManrajGrover) [inline share/goodie/cheat_sheets/json/pycharm.json]: Please make necessary change in all of the places.

File: share/goodie/cheat_sheets/json/pycharm.json [hunk 4]
```diff
@@ -0,0 +1,220 @@
...
+                "key": "[Shift] [F9]",
+                "val": "Debug"
+            }
+        ],
+        "Search/Replace": [
+            {
+                "key": "[Ctrl] [F]",
+                "val": "Find"
+            },            {
```
Comment (ManrajGrover) [inline share/goodie/cheat_sheets/json/pycharm.json]: Single space after `,`, not multiple. Please make necessary change everywhere.

File: share/goodie/cheat_sheets/json/pycharm.json [hunk 5]
```diff
@@ -0,0 +1,220 @@
+{
+    "id": "pycharm_cheat_sheet",
+    "name": "PyCharm",
+    "description": "Keyboard shortcuts for PyCharm. It's a python IDE, developed by JetBrains",
+
+    "metadata": {
+        "sourceName": "JetBrains",
+        "sourceUrl" : "https://www.jetbrains.com/help/pycharm/2016.3/keyboard-shortcuts-you-cannot-miss.html"
+    },
+
```
Comment (ManrajGrover) [inline share/goodie/cheat_sheets/json/pycharm.json]: Let's remove blank lines from source code, apart from one at the end.

File: share/goodie/cheat_sheets/json/pycharm.json (hunk 1 above)
Comment (ManrajGrover) [inline share/goodie/cheat_sheets/json/pycharm.json]: Let's reframe it to `Keyboard shortcuts for PyCharm, a Python IDE developed by JetBrains.`

File: share/goodie/cheat_sheets/json/pycharm.json [hunk 6]
```diff
@@ -0,0 +1,220 @@
...
+    "aliases": [
+         "pycharm ide", "python ide pycharm", "python jetbrains", "pychram editor", "ide python"
```
Comment (ManrajGrover) [inline share/goodie/cheat_sheets/json/pycharm.json]: Let's put one alias per line.

Comment (sahildua2305) [inline share/goodie/cheat_sheets/json/pycharm.json]: @Khakhulin `ide python` seems to be a too wide-scoped keyword to be used as an alias for this cheatsheet.

File: share/goodie/cheat_sheets/json/pycharm.json [hunk 7]
```diff
@@ -0,0 +1,265 @@
...
+  "metadata": {
+    "sourceName": "JetBrains",
+    "sourceUrl": "https://www.jetbrains.com/help/pycharm/2016.3/keyboard-shortcuts-you-cannot-miss.html"
+  },
+  "aliases": [
+    "pycharm ide",
+    "python ide pycharm",
+    "python jetbrains",
+    "pychram editor",
+    "ide python"
```
Comment (sahildua2305) [inline share/goodie/cheat_sheets/json/pycharm.json]: Hey @Khakhulin, I don't think `ide python` is a good alias for this IA. What do you think?

Comment (Khakhulin) [inline share/goodie/cheat_sheets/json/pycharm.json]: Hi @sahildua2305 ,
Why don't not? Yes, it's not precise name, but it's the most popular ide. Of course I can fix it.
Do you speak about indentation in the beginning? For instance, on 14 line.

**Comment by kirkins** [comment]
Seems all requested changes have been made and those LGTM.

Pycharm seems to be mis-spelled on the IA page but I don't think that relates to the code in pull request.

https://duck.co/ia/view/pychram_cheat_sheet

**Comment by mintsoft** [comment]
I'm going to close this for now (doing some house-keeping). Feel free to ping and I'll re-open if I've closed it in error

**Pull request closed without merging**